aten/src/ATen/cpu/vec256/vec256_bool.h

```diff
@@ -0,0 +1,331 @@
+      case 20:
+        return blend<1048575>(a, b);
+      case 21:
+        return blend<2097151>(a, b);
+      case 22:
+        return blend<4194303>(a, b);
+      case 23:
+        return blend<8388607>(a, b);
+      case 24:
+        return blend<16777215>(a, b);
+      case 25:
+        return blend<33554431>(a, b);
+      case 26:
+        return blend<67108863>(a, b);
+      case 27:
+        return blend<134217727>(a, b);
+      case 28:
+        return blend<268435455>(a, b);
+      case 29:
+        return blend<536870911>(a, b);
+      case 30:
+        return blend<1073741823>(a, b);
+      case 31:
+        return blend<2147483647>(a, b);
+    }
+    return b;
+  }
+  static Vec256<uint8_t> loadu(const void* ptr) {
+    return _mm256_loadu_si256(reinterpret_cast<const __m256i*>(ptr));
+  }
+  static Vec256<uint8_t> loadu(const void* ptr, uint8_t count) {
+    __at_align32__ uint8_t tmp_values[size()];
+    std::memcpy(tmp_values, ptr, count * sizeof(uint8_t));
+    return loadu(tmp_values);
+  }
+  void store(void* ptr, int count = size()) const {
+    if (count == size()) {
+      _mm256_storeu_si256(reinterpret_cast<__m256i*>(ptr), values);
+    } else if (count > 0) {
+      __at_align32__ uint8_t tmp_values[size()];
+      _mm256_storeu_si256(reinterpret_cast<__m256i*>(tmp_values), values);
+      std::memcpy(ptr, tmp_values, count * sizeof(uint8_t));
+    }
+  }
+  const uint8_t& operator[](int idx) const  = delete;
+  uint8_t& operator[](int idx)  = delete;
+  Vec256<uint8_t> abs() const {
+    return values;
+  }
+  Vec256<uint8_t> operator==(const Vec256<uint8_t>& other) const {
+    // _mm256_cmpeq_epi8 set all bits in byte to 1 if the two values equal. We
+    // only use the lowest bit in every byte to make sure it returns 1 instead
+    // of 255.
+    static auto one  = _mm256_set1_epi8(1);
+    return _mm256_and_si256(one, _mm256_cmpeq_epi8(values, other.values));
+  }
+  Vec256<uint8_t> operator!=(const Vec256<uint8_t>& other) const {
+    static auto one = _mm256_set1_epi8(1);
+    auto eq = _mm256_cmpeq_epi8(values, other.values);
+    return _mm256_and_si256(one, _mm256_xor_si256(one, eq));  // invert
+  }
+  Vec256<uint8_t> operator<(const Vec256<uint8_t>& other) const {
+    // cannot use _mm256_cmpgt_epi8 which is for signed bytes
+    static auto one = _mm256_set1_epi8(1);
+    auto min = _mm256_min_epu8(values, other.values);
+    auto gte = _mm256_cmpeq_epi8(min, other.values);
+    return _mm256_and_si256(one, _mm256_xor_si256(one, gte));  // invert
+  }
+  Vec256<uint8_t> operator<=(const Vec256<uint8_t>& other) const {
+    // cannot use _mm256_cmpgt_epi8 which is for signed bytes
+    static auto one = _mm256_set1_epi8(1);
+    auto min = _mm256_min_epu8(values, other.values);
+    return _mm256_and_si256(one, _mm256_cmpeq_epi8(min, values));
+  }
+  Vec256<uint8_t> operator>(const Vec256<uint8_t>& other) const {
+    // cannot use _mm256_cmpgt_epi8 which is for signed bytes
```

Comment:
do we need this now?

Reply:
I think yes. 

I think it is more natural/readable to use `_mm256_cmpgt_epi8` instead of `max`/`eq` operations. But, sadly AVX2 does not have a unsigned version for `>`.